### PR TITLE
Publish user manual in GitHub releases

### PR DESCRIPTION
**Pull request by 9999years**
~~Blocked on #250~~

Builds the mdBook user manual in Nix and uploads it to GitHub releases in a `.tar.xz`.

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -54,11 +54,11 @@
               # TODO: Bump the Rust version here...
               rustToolchain = final.pkgsBuildHost.rust-bin.stable."1.72.1".default.override {
                 targets =
-                  final.lib.optionals final.stdenv.isDarwin [
+                  final.lib.optionals final.stdenv.targetPlatform.isDarwin [
```
Comment (9999years) [inline flake.nix:57]: This should have always been `targetPlatform`, I think

File: nix/packages/ghciwatch.nix [hunk 2]
```diff
@@ -40,45 +40,58 @@
...
+              exit 1
+            fi
+          done
+        fi
+      '';
+    }
+    // (lib.optionalAttrs (stdenv.targetPlatform.isLinux && stdenv.targetPlatform.isx86_64) {
+      # Make sure we don't link with GNU libc so we can produce a static executable.
+      CARGO_BUILD_TARGET = "x86_64-unknown-linux-musl";
+    })
```
Comment (9999years) [inline nix/packages/ghciwatch.nix:89]: Moved these out of the final expression.

File: nix/packages/ghciwatch.nix [hunk 3]
```diff
@@ -90,6 +103,81 @@
       inherit cargoArtifacts;
     };
 
+  releaseArgs =
+    commonArgs
+    // {
+      # Don't run tests; we'll do that in a separate derivation.
+      # This will allow people to install and depend on `ghciwatch`
+      # without downloading a half dozen different versions of GHC.
+      doCheck = false;
```
Comment (Gabriella439) [inline nix/packages/ghciwatch.nix:112]: In principle if the build output doesn't include any references to these GHCs then they wouldn't be included as runtime dependencies.  Specifically, if `ghciwatch` is built, tested, and cached then pulling down the cache product for `ghciwatch` won't pull down the accompanying GHCs.

I don't feel that strongly about this, though.

File: nix/packages/ghciwatch.nix [hunk 4]
```diff
@@ -90,6 +103,81 @@
...
+
+  cli-markdown = stdenv.mkDerivation {
+    pname = "ghciwatch-cli-markdown";
+    inherit (commonArgs) version;
+
+    phases = ["buildPhase"];
+
+    nativeBuildInputs = [ghciwatch-with-clap-markdown];
+
+    buildPhase = ''
```
Comment (Gabriella439) [inline nix/packages/ghciwatch.nix:135]: Very minor suggestion:

```suggestion
    phases = ["installPhase"];

    nativeBuildInputs = [ghciwatch-with-clap-markdown];

    installPhase = ''
```

**Review by Gabriella439 — APPROVED** [review]

**Review by Gabriella439 — APPROVED** [review]

**Pull request merged**